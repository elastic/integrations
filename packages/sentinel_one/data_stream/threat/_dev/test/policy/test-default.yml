vars:
  url: http://host.tld
  api_token: test_api_token
  enable_request_tracer: false
data_stream:
  vars:
    initial_interval: 24h
    interval: 30s
    preserve_original_event: false
    tags:
      - forwarded
      - sentinel_one-threat
