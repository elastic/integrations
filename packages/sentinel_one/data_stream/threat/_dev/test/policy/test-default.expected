inputs:
    - data_stream:
        namespace: ep
      meta:
        package:
            name: sentinel_one
      name: test-default-sentinel_one
      streams:
        - config_version: 2
          cursor:
            last_update_at:
                value: '[[.last_event.threatInfo.updatedAt]]'
          data_stream:
            dataset: sentinel_one.threat
            type: logs
          interval: 30s
          publisher_pipeline.disable_host: true
          request.method: GET
          request.ssl: null
          request.tracer:
            enabled: false
            filename: ../../logs/httpjson/http-request-trace-*.ndjson
            maxbackups: 5
          request.transforms:
            - set:
                target: header.Authorization
                value: ApiToken ${SECRET_0}
            - set:
                target: url.params.limit
                value: "100"
            - set:
                target: url.params.sortBy
                value: updatedAt
            - set:
                target: url.params.sortOrder
                value: asc
            - set:
                default: '[[formatDate (now (parseDuration "-24h"))]]'
                target: url.params.updatedAt__gte
                value: '[[formatDate (parseDate .cursor.last_update_at)]]'
          request.url: http://host.tld/web/api/v2.1/threats
          response.pagination:
            - set:
                fail_on_template_error: true
                target: url.params.cursor
                value: '[[if (ne .last_response.body.pagination.nextCursor nil)]][[.last_response.body.pagination.nextCursor]][[end]]'
          response.split:
            ignore_empty_value: true
            target: body.data
          tags:
            - forwarded
            - sentinel_one-threat
      type: httpjson
      use_output: default
output_permissions:
    default:
        _elastic_agent_checks:
            cluster:
                - monitor
        _elastic_agent_monitoring:
            indices: []
        uuid-for-permissions-on-related-indices:
            indices:
                - names:
                    - logs-sentinel_one.threat-ep
                  privileges:
                    - auto_configure
                    - create_doc
secret_references:
    - {}
